pkg/kafka/encoding.go

```diff
@@ -188,3 +194,55 @@ func (d *Decoder) DecodeWithoutLabels(data []byte) (logproto.Stream, error) {
 func sovPush(x uint64) (n int) {
 	return (math_bits.Len64(x|1) + 6) / 7
 }
+
+// EncodeStreamMetadata encodes the stream metadata into a Kafka record
+// using the tenantID as the key and partition as the target partition
+func EncodeStreamMetadata(partition int32, topic string, tenantID string, streamHash uint64) *kgo.Record {
```

Comment:
@periklis why don't we change it to `(*kgo.Record, error)`? Then we can return an error on line `215`?

Reply:
Agree on that too, we should now hide the error